### PR TITLE
ioapi: add version 3.2

### PR DESCRIPTION
**Pull request by omsai**
New package for IOAPI.

So far only works on GNU/Linux.  

I've posted on the [mailing list](https://groups.google.com/forum/#!topic/spack/R6orWxLCtfg) about adding logic for other architectures supported by upstream, but I don't have exotic hardware to test anyway besides `linux x86_64`, so I figured I would send the working version with my architecture.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ioapi/package.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+from spack import *
+
+
+class Ioapi(MakefilePackage):
+    """Models-3/EDSS Input/Output Applications Programming Interface."""
+
+    homepage = "https://www.cmascenter.org/ioapi/"
+    url      = "https://www.cmascenter.org/ioapi/download/ioapi-3.2.tar.gz"
+    version('3.2', sha256=('56771ff0053d47f2445e00ff369bca7b'
+                           'fc484325a2816b2c648744e523134fe9'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ioapi/package.py]: You can put all of this on a single line. Spack exempts checksums from its line-length checks.

File: var/spack/repos/builtin/packages/ioapi/package.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        # The makefile uses stubborn assignments of = instead of ?= so
+        # edit the makefile instead of using environmental variables.
+        makefile = FileFilter('Makefile')
+        makefile.filter('^BASEDIR.*', 'BASEDIR = ' + self.build_directory)
+        makefile.filter('^INSTALL.*', 'INSTALL = ' + prefix)
+        makefile.filter('^BININST.*', 'BININST = ' + prefix.bin)
+        makefile.filter('^LIBINST.*', 'LIBINST = ' + prefix.lib)
+
+    def build(self, spec, prefix):
+        make()
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ioapi/package.py]: This is the default, no need to override `build`.

File: var/spack/repos/builtin/packages/ioapi/package.py (hunk 1 above)
Comment (omsai) [reply var/spack/repos/builtin/packages/ioapi/package.py]: Fixed in 31ff038

Out of curiosity, how does the spack CI, etc apply the exemption?  I thought it was from the `.flake8_packages` config file in the project root, but after making your suggested change, `flake8` now raises E501:

```console
$ flake8 --config=.flake8_packages var/spack/repos/builtin/packages/ioapi/package.py
var/spack/repos/builtin/packages/ioapi/package.py:9:1: F403 'from spack import *' used; unable to detect undefined names
var/spack/repos/builtin/packages/ioapi/package.py:17:80: E501 line too long (93 > 79 characters)
```

I have the latest version of that config file:

```console
$ GIT_PAGER= git log --oneline -n 1 .flake8_packages
a750237 flake8: add exceptions for overly pedantic camelcase rules from pep8-naming (#11477)
```

```console
$ flake8 --version
3.7.8 (mccabe: 0.6.1, pycodestyle: 2.5.0, pyflakes: 2.1.0) CPython 3.7.2 on Linux
```

File: var/spack/repos/builtin/packages/ioapi/package.py (hunk 2 above)
Comment (omsai) [reply var/spack/repos/builtin/packages/ioapi/package.py]: Fixed in 1060d09

File: var/spack/repos/builtin/packages/ioapi/package.py (hunk 1 above)
Comment (omsai) [reply var/spack/repos/builtin/packages/ioapi/package.py]: Ah, nevermind.  I read the Travis log and found it runs `./share/spack/qa/run-flake8-tests` which in turn runs `spack flake8`.  So I can simply run `spack flake8 path/to/my/package.py`

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/ioapi/package.py]: You can simply run `spack flake8` and Spack will find all files that have been changed in the current branch. Exemptions are listed [here](https://github.com/spack/spack/blob/develop/lib/spack/spack/cmd/flake8.py#L40,L84).

**Pull request merged**